Sources/Datadog/DatadogConfiguration.swift

```diff
@@ -401,6 +402,18 @@ extension Datadog {
                 return self
             }
 
+            /// Sets the custom mapper for `logEvent`. This can be used to modify logs before they are send to Datadog.
+            /// - Parameter mapper: the closure taking `logEvent` as input and expecting `logEvent` as output.
+            /// The implementation should obtain a mutable version of the `logEvent`, modify it and return it.
+            ///
+            /// **NOTE** The mapper intentionally prevents from returning a `nil` to drop the `logEvent` entirely, this ensures that all logs are sent to Datadog.
```

Comment:
We should check with Android SDK (cc @xgouchet ) but I think we don't need this limitation for `LogEvents` - any `LogEvent` can be safely dropped. 

This was only required for `RUMViewEvents` and `SpanEvents` which have hierarchical structure and dropping a particular event could break it (e.g. dropping a root `SpanEvent` would leave its child spans inconsistent; similar for dropping the `RUMViewEvent` which is referenced by some action / resource / error events).

That would mean, the mapper signature should be:
```swift
(LogEvent) -> LogEvent?
```
and we need to properly handle it downstream.